RecoEgamma/ElectronIdentification/plugins/ElectronHEEPIDValueMapProducer.cc

```diff
@@ -234,21 +232,42 @@ void ElectronHEEPIDValueMapProducer::produce(edm::Event& iEvent, const edm::Even
   auto ebRecHitHandle = getHandle(iEvent, ebRecHitToken_);
   auto eeRecHitHandle = getHandle(iEvent, eeRecHitToken_);
   auto beamSpotHandle = getHandle(iEvent, beamSpotToken_);
-  auto candHandles = getHandles(iEvent, candTokens_);
 
   bool isAOD = isEventAOD(iEvent, eleToken_);
   const auto& candVetos = isAOD ? candVetosAOD_ : candVetosMiniAOD_;
 
+  std::vector<EleTkIsolFromCands::PreselectedTracks> preselectedTracksForIso03;
+  std::vector<EleTkIsolFromCands::PreselectedTracks> preselectedTracksForIso04;
+
+  {
+    int handleNr = 0;
+    for (auto const& handle : getHandles(iEvent, candTokens_)) {
+      if (handle.isValid()) {
+        preselectedTracksForIso03.push_back(trkIsoCalc_.preselectTracks(*handle, candVetos[handleNr]));
```

Comment:
shouldn't there be a check somewhere that there are any electrons to consider?
This seems too heavy for somewhat empty events